src/Flysystem/Adapter/FedoraAdapter.php

```diff
@@ -93,9 +105,24 @@ public function read($path) {
    * {@inheritdoc}
    */
   public function readStream($path) {
-    $response = $this->fedora->getResource($path, ['Connection' => 'close']);
+    $headers = ['Connection' => 'close'];
+
+    // If the request is for a range
+    // pass that header to fedora.
+    if ($this->request && $this->request->headers->has('Range')) {
+      $range = $this->request->headers->get('Range');
+      if (str_starts_with($range, 'bytes=')) {
+        // Since \Symfony\Component\HttpFoundation\BinaryFileResponse seeks
+        // to the start of the range based on the request's Range header
+        // we need to always set start to 0 so fedora returns
+        // all the bytes between zero and the start of the range.
+        [$start, $end] = explode('-', substr($range, 6), 2) + [1 => ""];
+        $headers['Range'] = "bytes=0-$end";
```

Comment:
I'm not sure I understand what you are saying here. We have to grab from the start of the resource to whatever the end bytes are so we can't really chunk data from Fedora because the BinaryFileResponse needs to start at the beginning? Perhaps we should not be using the BinaryFileResponse and switch to a StreamingResponse then?

Reply:
FWIW I feel like the upstream repositories don't handle range requests like this well enough and the proper place to make these fixes is in those upstream repositories which we can then leverage.

Reply:
Ok, it would probably be good to document the issues in those upstream repositories somewhere in case there is an oppourtunity to correct them. For instance asking Flysystem to not assume we are backed by a file on disk. But you are correct that this isn't anything worse then we were already doing in my opinion.

Reply:
Agreed. I've been trying to decide the best place to either file an issue or put up a PR. Definitely hope to push this further for full range request support for streams.